assets/js/job-editor/mount.tsx

```diff
@@ -106,27 +111,32 @@ export default {
 
     checkAdaptorVersion(adaptor);
 
-    this.findWorkflowEditorStore().then(workflowStore => {
-      const job = workflowStore.getState().getById<Lightning.Job>(jobId);
-
-      if (!job) {
-        console.error('Job not found', jobId);
-        return;
-      }
+    pRetry(
+      async attempt => {
+        console.debug('JobEditor', { attempt });
 
-      if (JobEditorComponent) {
-        this.componentRoot?.render(
-          <JobEditorComponent
-            adaptor={adaptor}
-            source={job.body}
-            metadata={this.metadata}
-            disabled={disabled === 'true'}
-            disabledMessage={disabledMessage}
-            onSourceChanged={src => this.handleContentChange(src)}
-          />
-        );
-      }
-    });
+        const workflowStore = await this.findWorkflowEditorStore();
+        const job = workflowStore.getState().getById<Lightning.Job>(jobId);
+        if (!job) {
+          throw new Error('Job not found in store yet');
+        }
+        if (JobEditorComponent) {
+          this.componentRoot?.render(
+            <JobEditorComponent
+              adaptor={adaptor}
+              source={job.body}
+              metadata={this.metadata}
+              disabled={disabled === 'true'}
+              disabledMessage={disabledMessage}
+              onSourceChanged={src_1 => this.handleContentChange(src_1)}
+            />
+          );
+        } else {
+          throw new Error('JobEditorComponent not loaded yet');
+        }
+      },
+      { retries: 5 }
+    );
```

Comment:
@josephjclark going off my theory I mentioned earlier, I added this to the render cycle.. it's not a long term fix but thought it's worth a shot.